North-Seattle-College/ad440-winter2021-thursday-repo
# Caching Feature Flag for /users

**What this PR Does:**
- closes #204 
- minor edits to /automation/alert and /automation/vnet files
- changes /Automation to /automation
- adds CACHE_TOGGLE variable to determine if caching will be available

**To Test:**
- (1) in VSCode add the line `"CACHE_TOGGLE": "On"` to functions/local.settings.json
- (2) under the "Run/Debug" tab in VSCode, run the functions
- (3) go to Postman, c/p the endpoint for /users that VSCode will display on the terminal
- (4) run a GET 
  - you should see over 1000 results in Postman
  - VSCode terminal should display the following:
  - ![image](https://user-images.githubusercontent.com/57076614/109789615-5c2a9380-7bc5-11eb-9ee8-13cd0ab64e35.png)
  - (if not, skip to step 6)
- (5) run another GET
  - VSCode terminal should display the following:
  -  ![image](https://user-images.githubusercontent.com/57076614/109786242-d22cfb80-7bc1-11eb-8fc9-f438b6793c21.png)
- (6) run a POST with a body like so:
   ```
   {
   	"firstName": "First",
	 "lastName": "Last",
	"email": "email@email.com"
   }
   ``` 
   - VSCode terminal should display the following:
   - ![image](https://user-images.githubusercontent.com/57076614/109786543-1cae7800-7bc2-11eb-9d37-1bad0cdc4ab7.png)
- (7) run (yet) another GET
   - VSCode terminal should display message from (4) again

- (8) Change the CACHE_TOGGLE value in /functions/local.settings.json to "Off"
- (9) run a GET
   - VSCode terminal should display the following:
   - ![image](https://user-images.githubusercontent.com/57076614/109787074-b2e29e00-7bc2-11eb-82c3-2413ae0637d0.png)
- (10) run a POST with a body like in (6)
   - VSCode terminal should display the following:
   - ![image](https://user-images.githubusercontent.com/57076614/109787355-fd641a80-7bc2-11eb-924c-14a9b49ac2eb.png)
 - (11) run a last GET
    - VSCode terminal should display the same as in (9)
